src/peft/tuners/lora.py

```diff
@@ -630,22 +641,33 @@ def add_weighted_adapter(self, adapters, weights, adapter_name, combination_type
 
                 target_lora_A.data = target_lora_A.data * 0.0
                 target_lora_B.data = target_lora_B.data * 0.0
-                if combination_type == "linear":
+                if combination_type == "linear" or combination_type == "cat":
```

Comment:
More of a personal preference, but I would strictly separate the two combination types, even if it means a little bit of code is duplicated. So the "linear" combination can stay the same, just a new `elif` branch is added for "cat".

Reply:
Alright, fixed it!